src/pages/workspace/tags/WorkspaceViewTagsPage.tsx

```diff
@@ -222,11 +237,16 @@ function WorkspaceViewTagsPage({route}: WorkspaceViewTagsProps) {
         }
 
         if (enabledTagCount > 0) {
+            const tagsToDisableCount = Object.keys(tagsToDisable).length;
             options.push({
                 icon: Expensicons.Close,
                 text: translate(enabledTagCount === 1 ? 'workspace.tags.disableTag' : 'workspace.tags.disableTags'),
                 value: CONST.POLICY.BULK_ACTION_TYPES.DISABLE,
                 onSelected: () => {
+                    if (tagsToDisableCount === enabledTagCount && policy?.requiresTag) {
```

Comment:
`tagsToDisableCount` === `enabledTagCount` is always true. The `enabledTagCount` refers to the number of selected tags that are enabled and those are the same that we are going to disable.